From 71ef4e4b6a1ebef5e8a22e3a175a5e6ce16d8eed Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Sat, 19 Jan 2019 17:06:24 +0100
Subject: [PATCH 05/12] worker: allow negotiating AC-DTLS12 with openconnect

This doesn't have the anyconnect client bug with parsing the
server hello.

Signed-off-by: Nikos Mavrogiannopoulos <nmav@gnutls.org>
---
 src/worker-http.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/worker-http.c b/src/worker-http.c
index 963c476f..67a88602 100644
--- a/src/worker-http.c
+++ b/src/worker-http.c
@@ -441,8 +441,9 @@ void header_value_check(struct worker_st *ws, struct http_req_st *req)
 		 * anyconnect's openssl fail: https://gitlab.com/gnutls/gnutls/merge_requests/868
 		 */
 #ifdef gnutls_check_version_numeric
-		if (!gnutls_check_version_numeric(3,6,6) &&
-		    (!gnutls_check_version_numeric(3,3,0) || gnutls_check_version_numeric(3,6,0))) {
+		if (req->user_agent_type != AGENT_OPENCONNECT &&
+		    (!gnutls_check_version_numeric(3,6,6) &&
+		    (!gnutls_check_version_numeric(3,3,0) || gnutls_check_version_numeric(3,6,0)))) {
 			break;
 		}
 #endif
-- 
2.11.0

